pkg/types/powervs/validation/machinepool.go

```diff
@@ -0,0 +1,69 @@
+package validation
+
+import (
+	"math"
+	"regexp"
+	"strconv"
+
+	"github.com/google/uuid"
+	"github.com/openshift/installer/pkg/types/powervs"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateMachinePool checks that the specified machine pool is valid.
+func ValidateMachinePool(p *powervs.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Validate VolumeIDs
```

Comment:
Do you want to validate that the volume IDs are unique?

Reply:
yes, can we do that in a followon PR as well?